nest/core/decorators/injectable.py

```diff
@@ -1,25 +1,27 @@
 from injector import inject
 
 from nest.common.constants import DEPENDENCIES, INJECTABLE_NAME, INJECTABLE_TOKEN
-from nest.core.decorators.utils import get_instance_variables, parse_dependencies
-
-
-def Injectable(cls: str = None):
-    def __init__(self, *args, **kwargs):
-        ...
-
-    if "__init__" in cls.__dict__:
-        pass
-    else:
-        cls.__init__ = __init__
-
-    dependencies = parse_dependencies(cls)
-
-    setattr(cls, DEPENDENCIES, dependencies)
-
-    setattr(cls, INJECTABLE_TOKEN, True)
-    setattr(cls, INJECTABLE_NAME, cls.__name__)
-
-    inject(cls)
-
-    return cls
+from nest.core.decorators.utils import parse_dependencies
+
+
+def Injectable(cls=None, *args, **kwargs):
```

Comment:
what's the different between cls and inner_cls?

Reply:
### Technical summary

- `cls` This is the class that the Injectable decorator is being applied to if the decorator is used **without parentheses**. In other words, if you use the decorator like this:

```python
@Injectable
class MyClass:
    pass
```
In this case, `cls` will be `MyClass`.

- `inner_cls` This is always the class being decorated, **whether the decorator is called with parentheses or not**. When the decorator is called with parentheses, `inner_cls` will be the class passed to the decorator function.


### Why is this useful?
- Code Reusability: This pattern makes the Injectable decorator reusable and extensible without needing to rewrite or overload it for different use cases.
- Ease of Use: It provides a convenient way to apply dependency injection to classes with minimal syntax, making the code cleaner and more readable.
- Future-Proofing: Even if no arguments are needed now, the pattern allows for easy addition of functionality in the future without changing the way existing classes are decorated.

### Summary
This approach allows the Injectable decorator to be used in a flexible and future-proof manner, supporting both simple and potentially more complex usage scenarios while keeping the implementation clean and maintainable.

Reply:
ohh Now i get it, this is brilliant, and in the future, it will enable to add some variables to this decorator (like scope for example, if I want that provider/service will be singleton or other type of injection scope. this is very nice change.